### ISSUE TITLE
Definition of dynamic templates.

### ISSUE THREAD

**Issue by VKambulov** [babenkoivan/elastic-migrations#8]
Hello! I think it would be a good idea to make it possible to insert dynamic templates into migrations. For example, we have an object with a large nesting of related relations, the names of the fields and the purpose are approximately similar:
```json
{
  "id": 1,
  "name": "123456",
  "manufacturer": "carry-a-tune technologies",
  "SKU": 12345,
  "brand": {
    "id": 1,
    "name": "test",
    "main_manufacturer": "carry-a-tune technologies",
    "SKU": 12333,
    "organization": {
      "id": 1,
      "name": "Mayo Corporation",
      "SKU": 5544
    }
  }
}
```
In order not to assign a keyword type to each SKU field, we could do something like this:
```json
{
  "mappings": {
    "dynamic_templates": [
      {
        "manufacturer": {
          "match": "*manufacturer",
          "mapping": {
            "type": "keyword"
          }
        }
      },
      {
        "SKU": {
          "match": "SKU",
          "mapping": {
            "type": "keyword"
          }
        }
      },
      {
        "strings": {
          "match_mapping_type": "string",
          "mapping": {
            "type": "text",
            "fields": {
              "raw": {
                "type":  "keyword",
                "ignore_above": 256
              }
            }
          }
        }
      }
    ],
    "properties": {
      "id": {
        "type": "text"
      },
      "data": {
        "type": "nested"
      }
    }
  }
}
```

I have not found how to define dynamic templates in the current implementation, and judging by the code, this is not possible now, since the Mapping class works only with properties.
I would suggest adding the `dynamicTemplates` function and passing an array into it, or making dynamic templates define similar to the `__call` function, for example:
```php
$properties = [
 'match' => 'id',
 'mapping' => [
  'type' => 'keyword'
 ]
];

$mapping->template('name', $properties);
```
or:
```php
$maping->dynamicTemplates([
 'name' => [
   'match' => 'id',
   'mapping' => [
    'type' => 'keyword'
   ]
]);
```

In some situations, I think dynamic templates could help a lot.
Please correct me if I am wrong about something, or if this possibility already exists.

Updated: Sorry, I saw late that this is more related to Elastic Adapter.

**Comment by babenkoivan** [comment]
Hi @VKambulov, it's a good feature to add in the package. I've added it in my list.

**Comment by babenkoivan** [comment]
Hi @VKambulov, sorry for taking so long. The feature is now available in [v1.3.0](https://github.com/babenkoivan/elastic-migrations/releases/tag/v1.3.0).

```php
$mapping->dynamicTemplate('my_template_name', [
        'match_mapping_type' => 'long',
        'mapping' => [
            'type' => 'integer',
        ],
]);
```
